src/PackageVersions/Installer.php

```diff
@@ -80,27 +80,33 @@ public static function getSubscribedEvents()
      */
     public static function dumpVersionsClass(Event $composerEvent)
     {
-        $io = $composerEvent->getIO();
+        $composer = $composerEvent->getComposer();
+        $versions = iterator_to_array(self::getVersions($composer->getLocker(), $composer->getPackage()));
 
-        $io->write('<info>ocramius/package-versions:</info>  Generating version class...');
+        if (!isset($versions['ocramius/package-versions'])) {
+            //plugin must be globally installed - we only want to generate versions for projects which specifically
```

Comment:
@AydinHassan just a doubt, given that there's no real way to know from the code: what about when the package is indirectly depending on `package-versions`? Will it still run?